vterm.el

```diff
@@ -927,14 +927,20 @@ in README."
   "Get the position of the end of current prompt.
 More information see `vterm--prompt-tracking-enabled-p' and
 `Directory tracking and Prompt tracking'in README. "
-  (let ((end-point (vterm--get-end-of-line)))
+  (let ((end-point (vterm--get-end-of-line))
+        prompt-point)
     (save-excursion
-      (beginning-of-line)
-      (ignore-errors ;; if no point prompt, don't error out, stay at the beginning-of-line
-        (if vterm-copy-use-vterm-prompt
-            (goto-char (next-single-char-property-change (point) 'vterm-prompt nil end-point))
-          (search-forward-regexp vterm-copy-prompt-regexp end-point)))
-      (point))))
+      (if (and vterm-use-vterm-prompt
+               (vterm--prompt-tracking-enabled-p))
+          (if (get-text-property end-point 'vterm-prompt)
+              end-point
+            (setq prompt-point (previous-single-property-change end-point 'vterm-prompt))
+            (when prompt-point (setq prompt-point (1- prompt-point))))
+        (goto-char end-point)
+        (if (search-backward-regexp term-prompt-regexp (vterm--get-beginning-of-line) 'noerror)
```

Comment:
`vterm--get-prompt-point`:  Get the position of the end of current prompt. 
You should not  set the bound for `search-backward-regexp` .

`(vterm--at-prompt-p)` would return `t` when the point is at the beginning line ,this is not correct.